kernel/pmu_x86/src/lib.rs

```diff
@@ -690,7 +690,14 @@ pub fn start_samples(event_type: EventType, event_per_sample: u32, task_id: Opti
 
     sampled_events.sample_count = sample_count;
 
-    if event_per_sample > core::u32::MAX || event_per_sample <= core::u32::MIN {
+    if event_per_sample == 0 {
```

Comment:
@Ramla-I since you have more familiarity with this code, can you describe some of the constrains on this `event_per_sample` type? It's currently `u32` but I'm wondering if it might be more accurate to use an `i32` based on some docs from the ia64 manual. 

Please advise, thanks!

Reply:
Events_per_sample is the number of times the required event should occur before triggering an interrupt. I don't think making it an i32 makes sense, because you wouldn't want an event to occur a negative number of times. Though I agree the old check is excessive since the value is restrained by the type system anyway.

Each PMU counter value lies in the range [0, MAX_COUNT) where the MAX_COUNT is the maximum value that can be stored in the register depending on the bit width. We program an interrupt to be triggered when the value of the counter is equal to MAX_COUNT, and we set the starting value of the counter to be (MAX_COUNT - event_per_sample).

Is there somewhere in the manual you saw something different?

(There is actually an issue here because we are making the assumption that the bit width of the registers is 32, and actually that value should be taken from the CPUID register, but we can discuss that separately.)

Reply:
Thanks, yeah I agree that we should open a separate issue for this. I'll do that and cite the section of the manual that talks about this register.

Reply:
Resolving this after opening #539. This code will be rewritten anyway.